### PR TITLE
fix: Always update is_identified = true for first identify or alias user

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

For https://github.com/PostHog/posthog/issues/11873

This is step (1) of making alias and identify the same. This will also help us see which users could be negatively impacted if any (thanks to https://github.com/PostHog/posthog/pull/12120).

## Is there any user impact we should inform users about?

TLDR: No, assuming  these libs  automatically define `anon_distinct_id` and it's not something users provide.
Libs = [`web`, `js`, `posthog-node`, `posthog-android`, `posthog-ios`, `posthog-react-native`, `RudderLabs JavaScript SDK`, `Segment`]

#### Why?

The PR only changes `is_identified` updating, specifically making it `true` during `$create_alias` calls too.

`is_identified` is used in exactly one place: blocking merges during `$identify` calls.
Because we're making it `true` more often, we're definitely still safeguarding the things we were safeguarding before.

Potential problem: we'll block merges, that we weren't blocking before causing persons to not get merged & surprising users. Note that merges break persons-on-events, so those PostHog users have a broken experience anyway.

The blocking happens based on `oldPerson` being identified during `$identify` calls, i.e. based on `anon_distinct_id` person having `is_identified=true`.

(1) The only teams we care about are the ones who use both `$is_identify` and `$create_alias` (if they only use `$is_identify` then it's a no-op because we only changed what happens during alias calls, if they only use `$create_alias` then it's a no-op because `is_identified` is only used during identify calls).

(2) For those teams this would potentially change for folks who call identify, where currently oldPerson (passed as $anon_distinct_id) is not identified, we can look at the libraries used.

Looking up the data based on (1) and (2) https://metabase.posthog.net/question/445-mixed-usage-of-identify-and-alias

Over the past 30 days on cloud it that results in these libs [`web`, `js`, `posthog-node`, `posthog-android`, `posthog-ios`, `posthog-react-native`, `RudderLabs JavaScript SDK`, `Segment`].

I'm assuming these libs automatically define `anon_distinct_id` and it's not something users provide (that they could have used in `create_alias` previously).

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: plugin-server/tests/main/process-event.test.ts [hunk 1]
```diff
@@ -1711,7 +1711,7 @@ describe('when handling $create_alias', () => {
 
         // Make sure there is one non-identified person
```
Comment (macobo) [inline plugin-server/tests/main/process-event.test.ts:1712]: Example of why commenting tests can be problematic - this comment is now out of sync.

File: plugin-server/src/worker/ingestion/person-state.ts [hunk 2]
```diff
@@ -330,6 +330,11 @@ export class PersonState {
             return
         }
 
+        // The first distinct_id that is used in identify or alias (not the alias_id nor anon_id)
```
Comment (macobo) [inline plugin-server/src/worker/ingestion/person-state.ts:333]: This comment in this spot is not useful IMO - it adds more confusion than it resolves since it talks about things not relevant on the next lines.

I recommend removing it or e.g. documenting what is_identified column means elsewhere in the code.

File: plugin-server/src/worker/ingestion/person-state.ts [hunk 3]
```diff
@@ -330,6 +330,11 @@ export class PersonState {
...
+        // cannot be used as alias_id nor anon_id in the future.
```
Comment (macobo) [inline plugin-server/src/worker/ingestion/person-state.ts:334]: Nit: Let's use correct terminology - there's no such thing as `alias_id` or `anon_id`. Someone reading this comment will start grepping for these and not find anything, resulting in confusion.

**Comment by macobo** [comment]
Note: This is not a `fix` - this is a breaking change! Conventional commits do allow for tagging these via `BREAKING CHANGE:` tags.

Also this is missing tests - is this intended to be a draft, what are the todos?

**Comment by neilkakkar** [comment]
> TLDR: No, assuming these libs automatically define anon_distinct_id and it's not something users provide.

Bootstrapping allows defining the `anon_distinct_id` in `web` and `posthog-react-native`. Why does that change things?

Edit: ah, the final line, assuming this provided one is also used in `$create_alias`. Yeah, we're good then, overriding anonymous ID is different from overriding the distinctID.

`posthog-node` seems to be the only backend lib in the mix here, which also doesn't deal with anon IDs. Typo?

**Comment by tiina303** [comment]
> posthog-node seems to be the only backend lib in the mix here, which also doesn't deal with anon IDs. Typo?

I think that comes from us using js lib for node, which should be fine for the same reason js itself is fine 🤷‍♀️ 

> Also this is missing tests - is this intended to be a draft, what are the todos?

Tests aren't in this PR to avoid merge conflicts, because all of those would be in person-state.test.ts, which doesn't have any for alias atm and there's a separate PR that we're increasing the test coverage there.

**Comment by neilkakkar** [comment]
Just checked the linked query, this is a bug in `posthog-node`, it shouldn't be sending random anon distinct IDs.

https://github.com/PostHog/posthog-js-lite/issues/26

**Comment by tiina303** [comment]
> Note: This is not a fix - this is a breaking change! Conventional commits do allow for tagging these via BREAKING CHANGE: tags.

If you read the summary - this won't impact anyone currently, the breaking change will come later once we start restricting `$create_alias` calls.

**Review by macobo — APPROVED** [review]
